src/dataflow-types/src/types.rs

```diff
@@ -501,6 +502,36 @@ pub mod aws {
     #[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
     pub struct SerdeUri(#[serde(with = "http_serde::uri")] pub Uri);
 
+    fn any_serde_uri() -> impl Strategy<Value = SerdeUri> {
+        r"(http|https)://[a-z][a-z0-9]{0,10}/?([a-z0-9]{0,5}/?){0,3}"
+            .prop_map(|s| SerdeUri(s.parse().unwrap()))
+    }
```

Comment:
curious, how did you come up with this?

Reply:
`proptest` has this magic ability of being able to take regex patterns and automatically generate samples from them.
I just hand wrote a pattern that was always going to be a valid URI and called it a day: it's not perfect, but my main interest here was doing the Protobuf roundtripping assuming that we already have a valid URI as an input.

There's a [`proptest-http`](https://crates.io/crates/proptest-http) crate that includes strategies for generating `Uri`s, but it's barely used and adding a new dependency for something this minor felt like an overkill to me.

Reply:
yeah, I think your strategy here is good! Maybe add a comment explaining that we are just generating some generic random urls!

Reply:
Do you think a841d75106504a52a3447af5ba655b6e884b9ced explains it well enough?

Reply:
yes!